pkg/meta/cassandra/handler/grpc/handler.go

```diff
@@ -56,24 +75,79 @@ func (s *server) GetMeta(ctx context.Context, key *payload.Meta_Key) (*payload.M
 		switch {
 		case errors.IsErrCassandraNotFound(err):
 			log.Warnf("[GetMeta]\tnot found\t%v\t%s", key.GetKey(), err.Error())
+			err = status.WrapWithNotFound(fmt.Sprintf("GetMeta API: not found: key %s", key.GetKey()), err,
+				&errdetails.RequestInfo{
+					ServingData: errdetails.Serialize(key),
+				},
+				&errdetails.ResourceInfo{
+					ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/meta/Meta.GetMeta",
+					Owner:        errdetails.ValdResourceOwner,
+					Description:  err.Error(),
+				}, info.Get())
 			if span != nil {
 				span.SetStatus(trace.StatusCodeNotFound(err.Error()))
 			}
-			return nil, status.WrapWithNotFound(fmt.Sprintf("GetMeta API Cassandra key %s not found", key.GetKey()), err, info.Get())
-
+			return nil, err
 		case errors.IsErrCassandraUnavailable(err):
 			log.Warnf("[GetMeta]\tunavailable\t%+v", err)
+			err = status.WrapWithUnavailable(fmt.Sprintf("GetMeta API: Cassandra unavailable: key %s", key.GetKey()), err,
+				&errdetails.RequestInfo{
+					ServingData: errdetails.Serialize(key),
+				},
+				&errdetails.ResourceInfo{
+					ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/meta/Meta.GetMeta",
+					Owner:        errdetails.ValdResourceOwner,
+					Description:  err.Error(),
+				}, info.Get())
 			if span != nil {
 				span.SetStatus(trace.StatusCodeUnavailable(err.Error()))
 			}
-			return nil, status.WrapWithUnavailable("GetMeta API Cassandra unavailable", err, info.Get())
-
+			return nil, err
+		case isInternalError(err):
+			log.Warnf("[GetMeta]\tinternal error\t%+v", err)
+			err = status.WrapWithInternal(fmt.Sprintf("GetMeta API: internal error occurred: key %s", key.GetKey()), err,
+				&errdetails.RequestInfo{
+					ServingData: errdetails.Serialize(key),
+				},
+				&errdetails.ResourceInfo{
+					ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/meta/Meta.GetMeta",
+					Owner:        errdetails.ValdResourceOwner,
+					Description:  err.Error(),
+				}, info.Get())
+			if span != nil {
+				span.SetStatus(trace.StatusCodeInternal(err.Error()))
+			}
+			return nil, err
+		case isFailedPrecondition(err):
+			log.Warnf("[GetMeta]\tfailed precondition\t%+v", err)
+			err = status.WrapWithInternal(fmt.Sprintf("GetMeta API: failed precondition: key %s", key.GetKey()), err,
```

Comment:
maybe
```suggestion
			err = status.WrapWithFailedPreconditionl(fmt.Sprintf("GetMeta API: failed precondition: key %s", key.GetKey()), err,
```

Reply:
and how about add Precondition Error details?

Reply:
oops sorry...